packages/compass-connections/src/hooks/use-can-open-new-connections.ts

```diff
@@ -0,0 +1,69 @@
+import { useConnectionsManagerContext } from '../provider';
+import { ConnectionsManagerEvents } from '../connections-manager';
+
+import { useEffect, useState } from 'react';
+import { usePreference } from 'compass-preferences-model/provider';
+import { useAllSavedConnections } from './use-all-saved-connections';
+
+export type CanNotOpenConnectionReason = 'maximum-number-exceeded';
+
+export function useCanOpenNewConnections(): {
+  numberOfConnectionsOpen: number;
+  maximumNumberOfConnectionsOpen: number;
+  canOpenNewConnection: boolean;
+  canNotOpenReason?: CanNotOpenConnectionReason;
+} {
+  const connectionManager = useConnectionsManagerContext();
+
+  const { favorites, nonFavorites } = useAllSavedConnections();
+  const maximumNumberOfConnectionsOpen =
+    usePreference('userCanHaveMaximumNumberOfActiveConnections') ?? 1;
+
+  const [numberOfConnectionsOpen, setNumberOfConnectionsOpen] = useState(0);
+  const [canOpenNewConnection, setCanOpenNewConnection] = useState(false);
+  const [canNotOpenReason, setCanNotOpenReason] = useState<
+    CanNotOpenConnectionReason | undefined
+  >(undefined);
```

Comment:
Since you're starting with 0 connections open, it might be fine to start `canOpenNewConnection` with true and `canNotOpenReason` with undefined?

Reply:
Well, that could be an option. The main reason to start with a disabled state is to ensure that while we are synchronising and calculating the state the user can not open new connections if they shouldn't allowed to.

Reply:
That also sounds good. I will resolve this one.